os/pm/pm_idle.c

```diff
@@ -137,6 +137,12 @@ void pm_idle(void)
 						pmdbg("Sleep abort! CPU%d\n", cpu);
 						goto EXIT;
 					}
+
+					/* If the target cpu is already paused it will not be in idle thread, so PM should abort */
+					if (up_is_cpu_paused(cpu)) {
+						pmdbg("Sleep abort! CPU%d is paused\n", cpu);
+						goto EXIT;
+					}
```

Comment:
I think, `ASSERT(!up_is_cpu_paused(cpu))` is better.

enter_critical_section() is required for cpu pause
so if enter_critical_section and cpu pause are called and this checking here, the system is already corruped